### ISSUE TITLE
NullPointerException in DownloadManager$InternalHandler.onContentLengthChanged()

### ISSUE THREAD

**Issue by tuesda** [google/ExoPlayer#8078]
### [REQUIRED] Issue description
- there are some crashes on sentry which occurs in DownloadManager.InternalHandler.onContentLengthChanged()
- this crash is new on exoplayer 2.12.0, there is zero crash on exoplayer 2.11.7

### [REQUIRED] Reproduction steps
It is hard to reproduce for me on my device, but I found 600 times crash on sentry which occurs on user's devices

### [REQUIRED] Link to test content
There is no specific content for this issue, regarding of this crash's frequency, I guess this crash is not related with specific media content.

### [REQUIRED] A full bug report captured from the device

```
java.lang.NullPointerException: throw with null exception
    at com.google.android.exoplayer2.util.Assertions.checkNotNull(Assertions.java:1)
    at com.google.android.exoplayer2.offline.DownloadManager$InternalHandler.onContentLengthChanged(DownloadManager.java:2)
    at com.google.android.exoplayer2.offline.DownloadManager$InternalHandler.handleMessage(DownloadManager.java:6)
    at android.os.Handler.dispatchMessage(Handler.java:107)
    at android.os.Looper.loop(Looper.java:224)
    at android.os.HandlerThread.run(HandlerThread.java:67)
```

### [REQUIRED] Version of ExoPlayer being used
2.12.0 
There is zero of this crash on Exoplayer version below 2.12.0 (like 2.11.7

### [REQUIRED] Device(s) and version(s) of Android being used
A variety of kinds of devices are found, list as below
![image](https://user-images.githubusercontent.com/3896690/96085530-88ce5f00-0ef3-11eb-8f42-7da278e2fb73.png)

### Additional Info
almost crash occurs after DownloadService is destroyed

**Comment by marcbaechinger** [comment]
Thanks for reporting. I'll look into this asap.

**Comment by finneapps** [comment]
I am also seeing this crash on 2.12.1. In my case It seems to be happening when a download fails after resuming the app

**Comment by ojw28** [comment]
What type of content is being downloaded in these cases? Adaptive (i.e., HLS / DASH / SmoothStreaming) or progressive (i.e., regular media files like MP4 / MP3, etc)?

**Comment by ojw28** [comment]
Also, are you passing an `Executor` to the `DownloadManager` constructor, or using one of the constructors that doesn't take one? If you are passing one, which thread does it execute the downloads on?

**Comment by ojw28** [comment]
If the answers are (1) progressive media, and (2) using a separate executor, then I have figured out what's wrong. We will provide a fix shortly.

**Comment by ojw28** [comment]
The fix ref'd above will be included in the `2.12.2` release, which is due in the next couple of weeks.
